src/App.js

```diff
@@ -1,43 +1,82 @@
-import React from "react";
+import React, {useState, useEffect} from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import Footer from './components/Footer'
 
-class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
-    }
+const headers = { accept: "application/json" }
+let url;
+
+const LOADING_STATE = {
+  resting: "resting",
+  loaded: "Loaded",
+  loading: "Loading... please wait",
+  error: "Something went worng..."
+};
+
+function App(props) {
+  const [jokeData, setJokeData] = useState({ })
+  const [searchTerm, setSearchTerm] = useState("hat")
+  const [loadingState, setLoadingState] = useState(LOADING_STATE.resting)
+  
+
+  function handleChange(event) {
+    const {value} = event.target
+    setSearchTerm(value);
+    console.log("hello")
+  };
+
+  function handleSubmit(event){
+    event.preventDefault();
+    setLoadingState(LOADING_STATE.loading);
+    fetchJokes()
+  };
+
+  function checkInput(string) {
+    const regex = /^[a-zA-Z0-9_]+$/
+    regex.test(string) 
+      ? url = `https://icanhazdadjoke.com/search?term=${searchTerm}`
+      : url = `https://icanhazdadjoke.com/search`;
   }
 
-  async componentDidMount() {
-    const url = 'https://icanhazdadjoke.com/search?term="cat"';
-    const fetchUrl = fetch(url, {
-      method: "GET",
-      headers: { accept: "application/json" }
-    });
-    const response = await fetchUrl;
-    const myData = await response.json();
-    this.setState({ jokeData: { results: myData.results } });
+  async function fetchJokes() {
+    setLoadingState(LOADING_STATE.loading)
+    checkInput(searchTerm)
+    try {
+      const fetchUrl = fetch(url, {
+        method: "GET", headers: headers
+      });
+      const response = await fetchUrl;
+      const myData = await response.json();
+      setJokeData(myData)
+      setLoadingState(LOADING_STATE.loaded)
+    } catch (e) {
+        setLoadingState(LOADING_STATE.error)
+    }
   }
 
-  // This needs rendering inside a map function, then rendering in the render as <div>{jokeList}</div>
-  // <Joke key="[ENTER THE KEY HERE]" joke={this.state}/>
-  render() {
-    const jokeList = this.state.jokeData.results.map(item => (
-      <Joke key={item.id} joke={item.joke} />
-    ))
-    return (
-      <div className="App">
-        <PageHeader />
-        <div className="joke-wrapper">
-          {jokeList}
-        </div>
+  useEffect(() => {
+    fetchJokes()
+  }, [])
+
```

Comment:
I"m getting Lint errors here that I don't really undertand. It's suggesting I use `useCallback` function so that it will update if either one of the array items is updated. I kind of get why, but I don't understand the syntax to use here.